noir_stdlib/src/hash/mod.nr

```diff
@@ -62,9 +74,15 @@ fn pedersen_hash_with_separator<let N: u32>(input: [Field; N], separator: u32) -
 pub fn pedersen_hash<let N: u32>(input: [Field; N]) -> Field
 // docs:end:pedersen_hash
 {
-    pedersen_hash_with_separator(input, 0)
+    __pedersen_hash_with_separator(input, 0)
 }
 
+#[foreign(pedersen_hash)]
+fn __pedersen_hash_with_separator<let N: u32>(input: [Field; N], separator: u32) -> Field {}
+
+#[foreign(pedersen_commitment)]
+fn __pedersen_commitment_with_separator<let N: u32>(input: [Field; N], separator: u32) -> [Field; 2] {}
+
```

Comment:
This is adding back the blackboxes since the noir-implemented version generated issues in rollup_base and public_kernel_tail (BB just would hang when asked about VKs, gate counts, etc) of it

Reply:
Opened an issue that might be the culprit https://github.com/noir-lang/noir/issues/5633#issue-2437837776